app/controllers/common/RemoveMemberController.scala

```diff
@@ -0,0 +1,115 @@
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.common.RemoveMemberView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RemoveMemberController @Inject()(
+                                        val controllerComponents: MessagesControllerComponents,
+                                        identify: IdentifierAction,
+                                        getData: DataRetrievalAction,
+                                        requireData: DataRequiredAction,
+                                        userAnswersCacheConnector: UserAnswersCacheConnector,
+                                        formProvider: RemoveMemberFormProvider,
+                                        view: RemoveMemberView
+                                      )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+
+
+  def onPageLoad(waypoints: Waypoints, eventType: EventType, index: Index): Action[AnyContent] =
+    (identify andThen getData(eventType) andThen requireData) { implicit request =>
+
+      val form = formProvider(eventTypeMessage(eventType), fullNameAtIndex(request.userAnswers, eventType, index))
+      val preparedForm = request.userAnswers.get(RemoveMemberPage(eventType, index)).fold(form)(form.fill)
+      Ok(view(preparedForm, waypoints, eventType, eventTypeMessage(eventType), index, fullNameAtIndex(request.userAnswers, eventType, index)))
+    }
+
+  def onSubmit(waypoints: Waypoints, eventType: EventType, index: Index): Action[AnyContent] = (identify andThen getData(eventType) andThen requireData).async {
+    implicit request =>
+      val form = formProvider(eventTypeMessage(eventType), fullNameAtIndex(request.userAnswers, eventType, index))
+
+      form.bindFromRequest().fold(
+        formWithErrors =>
+          Future.successful(BadRequest(
+            view(formWithErrors, waypoints, eventType, eventTypeMessage(eventType), index, fullNameAtIndex(request.userAnswers, eventType, index)))
+          ),
+        value => {
+          val maybeUserAnswersWithRemovedMember: UserAnswers = if (value) {
+            eventType match {
+              case Event1 => request.userAnswers.removeWithPath(MembersOrEmployersPage(eventType).apply(index))
+              case _ => request.userAnswers.removeWithPath(MembersPage(eventType).apply(index))
+            }
+          } else {
+            request.userAnswers
+          }
+          val updatedAnswers = maybeUserAnswersWithRemovedMember.setOrException(RemoveMemberPage(eventType, index), value)
+          userAnswersCacheConnector.save(request.pstr, eventType, updatedAnswers).map { _ =>
+            Redirect(RemoveMemberPage(eventType, index).navigate(waypoints, maybeUserAnswersWithRemovedMember, updatedAnswers).route)
+          }
+        }
+      )
+  }
+}
+
+object RemoveMemberController {
+  def eventTypeMessage(eventType: EventType): String = {
```

Comment:
This would be better on the EventType companion object I think. Also, it should read from messages file